### PR TITLE
scripts: dts: generate array values as initializer lists

### PR DESCRIPTION
**Pull request by pabigot**
Simplify use of property values that have multiple elements.

Signed-off-by: Peter A. Bigot <pab@pabigot.com>

Addresses the behavioral part of #20172

### CODE REVIEW

File: scripts/dts/gen_defines.py [hunk 1]
```diff
@@ -188,8 +188,12 @@ def write_props(node):
         elif prop.type == "int":
             out_dev(node, ident, prop.val)
         elif prop.type == "array":
+            initializer_vals = []
             for i, val in enumerate(prop.val):
+                initializer_vals.append(val)
                 out_dev(node, "{}_{}".format(ident, i), val)
+            out_dev(node, ident,
+                    "{" + ", ".join(map(str, initializer_vals)) + "}")
```
Comment (ulfalizer) [inline scripts/dts/gen_defines.py]: Could get rid of `initializer_vals` and do

```python
out_dev(node, ident,
        "{" + ", ".join(map(str, prop.val)) + "}")
```

It's already a list.

**Review by ulfalizer — CHANGES_REQUESTED** [review]

**Review by ulfalizer — APPROVED** [review]
Fine with me. Someone else could comment on how helpful it is in the code.

**Comment by galak** [comment]
What's an example of a dt property that would cause this generation?

**Comment by pabigot** [comment]
> What's an example of a dt property that would cause this generation?

@galak See #20172.

More generally, anywhere you'd use `int32-array` or similar DT Schema types.  It's specifically useful where the number of elements isn't specified, because then you can do things like:

    static const int elts[] = DT_COMPAT_ELTS;

**Comment by galak** [comment]
Can we add the example from #20172 or something similar to the commit message.  I'm good with the change, just feel like having a bit more context or something when we look back would be useful.

**Review by galak — CHANGES_REQUESTED** [review]
Can we add a bit more detail/context/example to the commit message.  Otherwise the change looks good.  [just want to have a bit more history when we look back in git to understand what this change was for]

**Comment by pabigot** [comment]
> Can we add a bit more detail/context/example to the commit message.

See update:
```
Simplify use of property values that have multiple elements by
providing all of them in an initializer list, in addition to each one
as an enumerated value.

For example if a driver requires a sequence of operations with
instance-specific delays between stages the durations can be specified
with:

    delays = <30000 20 45000>;

and the driver can use:

    static u32_t delays[] = DT_INST_0_COMPAT_DELAYS;

rather than enumerating the instances.  This is particularly useful
when the number of entries in the array varies per instance, in which
case such an initialization is not easily expressed in code.
```

**Review by galak — APPROVED** [review]

**Pull request merged**